### PR TITLE
Introduce USDT trace points

### PR DESCRIPTION
**Pull request by erthalion**
## Description

Add systemtap into the builder to allow defining USDT, and add one to capture events when a new process was discovered.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Updated documentation accordingly

**Automated testing**
  - [ ] Added unit tests
  - [ ] Added integration tests
  - [ ] Added regression tests

If any of these don't apply, please comment below.

## Testing Performed

Manual testing.

### CODE REVIEW

**Review by Stringy — APPROVED** [review]
LGTM! I can only hope that this will mean we get USDT probes _everywhere_ 😄

File: collector/lib/ProcessSignalHandler.cpp [hunk 1]
```diff
@@ -33,6 +35,11 @@ bool ProcessSignalHandler::Stop() {
 
 SignalHandler::Result ProcessSignalHandler::HandleSignal(sinsp_evt* evt) {
   const auto* signal_msg = formatter_.ToProtoMessage(evt);
+  const char* name = signal_msg->signal().process_signal().name().c_str();
```
Comment (Stringy) [inline collector/lib/ProcessSignalHandler.cpp:38]: now that I look again, I think we want to put this after line 43 (after we've checked the `signal_msg` pointer)

File: collector/lib/ProcessSignalHandler.cpp [hunk 2]
```diff
@@ -33,6 +35,11 @@ bool ProcessSignalHandler::Stop() {
...
+  const int pid = signal_msg->signal().process_signal().pid();
+
+  DTRACE_PROBE2(collector, process_signal_handler, name, pid);
+
   if (!signal_msg) {
```
Comment (ovalenti) [inline collector/lib/ProcessSignalHandler.cpp:43]: This test suggests that signal_msg may sometimes be NULL.

**Review by ovalenti — APPROVED** [review]
Looks good :+1:

**Comment by robbycochran** [comment]
Will this be part of release builds? Do you need to make any changes to midstream or konflux environments?

**Comment by erthalion** [comment]
> Will this be part of release builds? Do you need to make any changes to midstream or konflux environments?

Indeed, Konflux is failing now. From what I see it doesn't use the builder image, this should fix it: https://github.com/stackrox/collector/pull/1646

**Pull request merged**